make/photon/db/postgresql/registry.sql

```diff
@@ -0,0 +1,348 @@
+CREATE DATABASE registry ENCODING 'UTF8';
+
+\c registry;
+
+create table access (
+ access_id SERIAL PRIMARY KEY NOT NULL,
+ access_code char(1),
+ comment varchar (30)
+);
+
+insert into access (access_code, comment) values 
+('M', 'Management access for project'),
+('R', 'Read access for project'),
+('W', 'Write access for project'),
+('D', 'Delete access for project'),
+('S', 'Search access for project');
+
+create table role (
+ role_id SERIAL PRIMARY KEY NOT NULL,
+ role_mask int DEFAULT 0 NOT NULL,
+ role_code varchar(20),
+ name varchar (20)
+);
+
+/*
+role mask is used for future enhancement when a project member can have multi-roles
+currently set to 0
+*/
+
+insert into role (role_code, name) values 
+('MDRWS', 'projectAdmin'),
+('RWS', 'developer'),
+('RS', 'guest');
+
+create table harbor_user (
+ user_id SERIAL PRIMARY KEY NOT NULL,
+ username varchar(255),
+ email varchar(255),
+ password varchar(40) NOT NULL,
+ realname varchar (255) NOT NULL,
+ comment varchar (30),
+ deleted boolean DEFAULT false NOT NULL,
+ reset_uuid varchar(40) DEFAULT NULL,
+ salt varchar(40) DEFAULT NULL,
+ sysadmin_flag boolean DEFAULT false NOT NULL,
+ creation_time timestamp(0),
+ update_time timestamp(0),
+ UNIQUE (username),
+ UNIQUE (email)
+);
+
+insert into harbor_user (username, email, password, realname, comment, deleted, sysadmin_flag, creation_time, update_time) values 
+('admin', 'admin@example.com', '', 'system admin', 'admin user',false, true, NOW(), NOW()),
+('anonymous', 'anonymous@example.com', '', 'anonymous user', 'anonymous user', true, false, NOW(), NOW());
+
+create table project (
+ project_id SERIAL PRIMARY KEY NOT NULL,
+ owner_id int NOT NULL,
+ /*
+ The max length of name controlled by API is 30, 
+ and 11 is reserved for marking the deleted project.
+ */
+ name varchar (255) NOT NULL,
+ creation_time timestamp,
+ update_time timestamp,
+ deleted boolean DEFAULT false NOT NULL,
+ FOREIGN KEY (owner_id) REFERENCES harbor_user(user_id),
+ UNIQUE (name)
+);
+
+insert into project (owner_id, name, creation_time, update_time) values 
+(1, 'library', NOW(), NOW());
+
+create table project_member (
+ id SERIAL NOT NULL,
+ project_id int NOT NULL,
+ entity_id int NOT NULL,
+ /*
+ entity_type indicates the type of member,
+ u for user, g for user group
+ */
+ entity_type char(1) NOT NULL,
+ role int NOT NULL,
+ creation_time timestamp default 'now'::timestamp,
+ update_time timestamp default 'now'::timestamp,
+ PRIMARY KEY (id),
+ CONSTRAINT unique_project_entity_type UNIQUE (project_id, entity_id, entity_type)
+);
+
+CREATE FUNCTION update_update_time_at_column() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+  BEGIN
+    NEW.update_time = NOW();
+    RETURN NEW;
+  END;
+$$;
+
+CREATE TRIGGER project_member_update_time_at_modtime BEFORE UPDATE ON project_member FOR EACH ROW EXECUTE PROCEDURE update_update_time_at_column();
```

Comment:
Please do not create trigger in database, because it may cause performance issue and not easy to migrate in future

Reply:
Is there any other way to update the time column? I followed the document of postgresql.